src/main/java/com/example/onlinebookstore/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,21 @@
+package com.example.onlinebookstore.security;
+
+import com.example.onlinebookstore.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        return userRepository.findByEmail(username)
+                .orElseThrow(() -> new UsernameNotFoundException(
```

Comment:
do you handle this exception?

or it is handled by Spring?

Reply:
i'm not sure.
Should I try replacing orElseThrow ?

Reply:
you should send a request with not existing email and check it

Reply:
I don't fully understand what needs to be changed))) But I noticed that when the user is not valid, a 403 is thrown regarding access, and not that the user was not found